src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1148,20 +1141,34 @@ InferenceProfiler::SummarizeLatency(
     summary.stabilizing_latency_ns = summary.client_stats.avg_latency_ns;
   }
 
-  // calculate standard deviation
-  uint64_t expected_square_latency_us =
-      tol_square_latency_us / latencies.size();
-  uint64_t square_avg_latency_us = (summary.client_stats.avg_latency_ns *
-                                    summary.client_stats.avg_latency_ns) /
-                                   (1000 * 1000);
-  uint64_t var_us = (expected_square_latency_us > square_avg_latency_us)
-                        ? (expected_square_latency_us - square_avg_latency_us)
-                        : 0;
-  summary.client_stats.std_us = (uint64_t)(sqrt(var_us));
-
   return cb::Error::Success;
 }
 
+std::tuple<uint64_t, uint64_t>
+InferenceProfiler::GetMeanAndStdDev(const std::vector<uint64_t>& latencies)
+{
+  uint64_t avg_latency_ns{0};
+  uint64_t std_dev_latency_us{0};
+
+  // calculate mean of latencies
+  uint64_t tol_latency_ns{
+      std::accumulate(latencies.begin(), latencies.end(), 0ULL)};
```

Comment:
We might need to improve the average calculation here too since it is susceptible to overflows but it can be part of a future ticket.

Reply:
Latencies are nanoseconds. There's no realistic way the sum of latencies would exceed `UINT64_MAX`. That'd be hundreds of years of latencies added up.